lib/rester.rb

```diff
@@ -9,7 +11,22 @@ module Rester
   autoload(:Utils,       'rester/utils')
   autoload(:Middleware,  'rester/middleware')
 
+  @_request_infos ||= ThreadSafe::Cache.new
+
+  # Set up the Client middleware if it's a Rails application
+  if defined?(Rails) && Rails.application
+    Rails.configuration.middleware.use(Client::Middleware::RequestHandler)
+  end
```

Comment:
Is the assumption that if `Rails.application` is defined then `Rails.configuration.middleware` will also be defined?

Should we check if `Rails.configuration` and `Rails.configuration.middleware` are defined?